e2e/e2e-starter.sh

```diff
@@ -42,10 +53,24 @@ fi
 if [ "$create_cluster" == "true" ]; then
     log "Creating cluster $CLUSTER_NAME"
     clusterCreateStartTime=$(date +%s)
+    retval=0
     if [[ "$RESOURCE_GROUP_NAME" == *"windows"* ]]; then
-        az aks create -g $RESOURCE_GROUP_NAME -n $CLUSTER_NAME --node-count 1 --generate-ssh-keys --network-plugin azure -ojson
+        az aks create -g $RESOURCE_GROUP_NAME -n $CLUSTER_NAME --node-count 1 --generate-ssh-keys --network-plugin azure -ojson || retval=$?
     else
-        az aks create -g $RESOURCE_GROUP_NAME -n $CLUSTER_NAME --node-count 1 --generate-ssh-keys --network-plugin kubenet -ojson
+        az aks create -g $RESOURCE_GROUP_NAME -n $CLUSTER_NAME --node-count 1 --generate-ssh-keys --network-plugin kubenet -ojson || retval=$?
+    fi
+
+    if [ "$retval" -ne 0  ]; then
+        log "Other pipelines may be creating cluster $CLUSTER_NAME, waiting for ready"
```

Comment:
would this happen since it seems like that only the pipeline creating the cluster successfully will set create_cluster to true?

Reply:
Yes. Even the pipeline didn't find any cluster at first, it could meet conflict when trying to create a cluster.